sql/parser/builtins.go

```diff
@@ -761,3 +774,12 @@ func datumToRawString(datum Datum) (string, error) {
 
 	return "", fmt.Errorf("argument type unsupported: %s", datum.Type())
 }
+
+func round(x float64, n int64) (Datum, error) {
+	if n < 0 || n > 50 {
```

Comment:
I made the same mistake earlier - the fact that `float64` has 17 significant decimal digits is unrelated to the limits of `round()`. I'm having a hard time thinking of a use case for `round()` on such tiny numbers, but `round(1.2345e-100, 102)` is mathematically valid and equals `1.23e-100`.

The real limit for `round()` is the 11-bit exponent, which equals 308 decimal digits, and even then I think higher values should be allowed (unless strconv.FormatFloat chokes on them). A double simply can't represent anything smaller than that, so "rounding" a double to 500 digits is just the original number.